Helper/Config.php

```diff
@@ -163,42 +161,79 @@ public function getEnvironmentVariable(string $envVarName): ?string
-    public function isMerchantActive(): bool
+    public function isMerchantActive(?int $storeId): bool
     {
-        return (bool) $this->scopeConfig->getValue(self::IS_MERCHANT_ACTIVE);
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->isActive : false;
     }
 
     /**
-     * Retrieve an access token.
-     *
-     * @return string|null The NS8 Protect Access Token.
+     * Retrieve an access token for a given store.
+     * @param string $storeId ID of the store to retrieve token for
+     * @return string|null The NS8 Protect Access Token for the provided store.
+     */
+    public function getAccessToken(?int $storeId): ?string
+    {
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->token : null;
+    }
+
+    public function setAccessToken(?int $storeId, string $token): void
+    {
+        $metadata = new ProtectMetadata($token, $this->isMerchantActive($storeId));
+        $this->setStoreMetadata($storeId, $metadata);
+    }
+
+    /**
+     * @return ProtectMetadata
+     */
+    public function getStoreMetadata(?int $storeId): ?\stdClass
+    {
+        $metadatas = $this->getStoreMetadatas();
+        $storeId = $storeId !== null ? $storeId : self::EMPTY_STORE_ID;
+        return isset($metadatas[$storeId]) ? $metadatas[$storeId] : null;
+    }
+
+    /**
+     * Get all access tokens, for all stores.
+     * Structured as { [storeId: string]: ProtectMetadata };
+     * @return ProtectMetadata[]
      */
-    public function getAccessToken(): ?string
+    public function getStoreMetadatas(): array
     {
-        return $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $rawTokensJson = $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $accessTokens = (array)json_decode($rawTokensJson);
+        return $accessTokens === null ? [] : $accessTokens;
     }
 
     /**
-     * Saves a Magento configurable value in an encrypted format
+     * Stores an NS8 Protect access token for a given store.
      *
-     * @param string $key The key of the configurable value
-     * @param string $value The value we want to store and encrypt for the configurable value
+     * @param string $storeId ID of the store to store token for
+     * @param string $token The access token to store.
      * @return void
      */
-    public function setEncryptedConfigValue(string $key, string $value): void
+    public function setStoreMetadata(?int $storeId, ProtectMetadata $metadata): void
     {
-        $this->scopeWriter->save($key, $this->encryptor->encrypt($value));
+        $storeId = $storeId !== null ? $storeId : self::EMPTY_STORE_ID;
+        $accessTokens = $this->getStoreMetadatas();
+        $accessTokens[$storeId] = $metadata;
+        $this->scopeWriter->save(self::ACCESS_TOKEN_CONFIG_KEY, $this->encryptor->encrypt(json_encode($accessTokens)));
```

Comment:
What's the benefit of storing this as an array? It seems like if we're storing each access token to a store, we should set the scope at store-level and allow each one individually. That also prevents access across stores if permissions are restricted.

Reply:
I thought the same, but we decided in planning to use an array: https://app.clubhouse.io/ns8/story/36886/store-access-tokens-separately-by-shop

Reply:
Thanks for clarifying 👍!